test/spec.html

```diff
@@ -8,28 +8,146 @@
     <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
     <script src="../../web-component-tester/browser.js"></script>
 
-    <link rel="import" href="../juicy-element.html">
+    <link rel="import" href="../slot-all.html">
+    <script>
+        customElements.define('shadow-host', class extends HTMLElement{
+            constructor(){
+                super();
+                this.attachShadow({mode: 'open'}).innerHTML = `<div id="scope"><slot-all></slot-all></div>`;
+            }
+        });
+    </script>
 </head>
 
 <body>
-    <test-fixture id="element">
+    <test-fixture id="shadow-host">
         <template>
-            <juicy-element></juicy-element>
+            <shadow-host>
+                <div slot="name">Div slotted by name</div>
+                <div slot="name">Div2 slotted by name</div>
+                <div slot="name2">Div slotted by name2</div>
+                <div>default</div>
+            </shadow-host>
         </template>
     </test-fixture>
 </body>
 <script>
-    var juicyElement;
-    describe('juicy-element', function() {
-
-        beforeEach(function() {
-            juicyElement = fixture('element');
+    describe('slot-all when used in shadow root', function() {
+        var host;
+        beforeEach(function(done) {
+            host = fixture('shadow-host');
+            // Ensure Shadow DOM distribution
+            flush(done);
         });
 
         it('should be upgraded element', function() {
-            expect(customElements.get('juicy-element')).to.be.defined;
-            expect(juicyElement).to.be.an.instanceOf(customElements.get('juicy-element'));
+            expect(customElements.get('slot-all')).to.be.defined;
+            expect(host.shadowRoot.querySelector('slot-all')).to.be.an.instanceOf(customElements.get('slot-all'));
+        });
+
+        it('should create `slot` elements for all slot names used by host\'s children, and `all-slotted` attribute to them', function() {
+            debugger
+            expect(host.shadowRoot.querySelector(`#scope > slot[name][all-slotted]`)).to.be.not.null;
+            expect(host.shadowRoot.querySelectorAll(`#scope > slot[name][all-slotted]`)).to.have.length(2);
+            Array.from(host.children).forEach((child)=>{
+                child.hasAttribute('slot') &&
+                expect(host.shadowRoot.querySelector(`#scope > slot[name="${child.getAttribute('slot')}"][all-slotted]`)).to.be.not.null;
```

Comment:
Hint: another option would be to check `child.assignedSlot`. Not a blocker though.